src/httpserver.h

```diff
@@ -14,8 +14,17 @@ namespace util {
 class SignalInterrupt;
 } // namespace util
 
-static const int DEFAULT_HTTP_THREADS=4;
-static const int DEFAULT_HTTP_WORKQUEUE=16;
+/**
+ * The default value for `-rpcthreads`. This number of threads will be created at startup.
+ */
+static const int DEFAULT_HTTP_THREADS=16;
```

Comment:
Maybe we should just  make the default depend on how many cores are available? e.g. default: `num cores / 2`

Reply:
From the linked issue, the user is saying the default of 4 is not enough for a raspi, which has 4 cores. So this would lower that number on a raspi.

Reply:
I see, at least something based on the users system instead of picking a random number?

Reply:
Just number of cores would make sense as a default. At least it wouldn't be lower on raspis.

Reply:
i've always been kind of hestitant about adapting these values to the system automatically; it doesn't only depend on the CPU, but also the expected usage.

For a "fair" work queue it would indeed make sense to pick the number of CPU cores. However, bitcoin core's RPC threads often sleep for various reasons. Eg waiting for I/O, waiting for a lock in another thread, etc. So it's fine to have it a bit higher than the number of cores.
i'd definitely cap the minimum at 4, even for single-core CPUs.

i'm also not sure people running bitcoin core on say, giant 64+-core systems necessarily want 64+ RPC threads. They might be running it on the background as one of many processes, not intending to do a lot of parallel processing with it at all.